tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1647 @@
+        assert(*b == CharType{'H'});
+
+        const auto cb = literal_constructed.cbegin();
+        static_assert(is_same_v<remove_const_t<decltype(cb)>, typename str::const_iterator>);
+        assert(*cb == CharType{'H'});
+
+        const auto cb2 = const_literal_constructed.begin();
+        static_assert(is_same_v<remove_const_t<decltype(cb2)>, typename str::const_iterator>);
+        assert(*cb2 == CharType{'H'});
+
+        const auto e = literal_constructed.end();
+        static_assert(is_same_v<remove_const_t<decltype(e)>, typename str::iterator>);
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) // TRANSITION, VSO-1275530
+        assert(*prev(e) == CharType{'s'});
+#endif // defined(MSVC_INTERNAL_TESTING) || defined(__EDG__)
+
+        const auto ce = literal_constructed.cend();
+        static_assert(is_same_v<remove_const_t<decltype(ce)>, typename str::const_iterator>);
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) // TRANSITION, VSO-1275530
+        assert(*prev(ce) == CharType{'s'});
+#endif // defined(MSVC_INTERNAL_TESTING) || defined(__EDG__)
+
+        const auto ce2 = const_literal_constructed.end();
+        static_assert(is_same_v<remove_const_t<decltype(ce2)>, typename str::const_iterator>);
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) // TRANSITION, VSO-1275530
+        assert(*prev(ce2) == CharType{'s'});
+#endif // defined(MSVC_INTERNAL_TESTING) || defined(__EDG__)
+
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) || _ITERATOR_DEBUG_LEVEL != 2 // TRANSITION, 16.10p1
+        const auto rb = literal_constructed.rbegin();
+        static_assert(is_same_v<remove_const_t<decltype(rb)>, reverse_iterator<typename str::iterator>>);
+        assert(*rb == CharType{'s'});
+
+        const auto crb = literal_constructed.crbegin();
+        static_assert(is_same_v<remove_const_t<decltype(crb)>, reverse_iterator<typename str::const_iterator>>);
+        assert(*crb == CharType{'s'});
+
+        const auto crb2 = const_literal_constructed.rbegin();
+        static_assert(is_same_v<remove_const_t<decltype(crb2)>, reverse_iterator<typename str::const_iterator>>);
+        assert(*crb2 == CharType{'s'});
+
+        const auto re = literal_constructed.rend();
+        static_assert(is_same_v<remove_const_t<decltype(rb)>, reverse_iterator<typename str::iterator>>);
+        assert(*prev(re) == CharType{'H'});
+
+        const auto cre = literal_constructed.crend();
+        static_assert(is_same_v<remove_const_t<decltype(cre)>, reverse_iterator<typename str::const_iterator>>);
+        assert(*prev(cre) == CharType{'H'});
+
+        const auto cre2 = const_literal_constructed.rend();
+        static_assert(is_same_v<remove_const_t<decltype(cre2)>, reverse_iterator<typename str::const_iterator>>);
+        assert(*prev(cre2) == CharType{'H'});
+#endif // defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) || _ITERATOR_DEBUG_LEVEL != 2
+    }
+
+    { // capacity

```

Comment:
I don't see `basic_string::length()` being tested anywhere - are we missing test coverage?